### DISCUSSION TITLE
How does it compare to Bazel?

### DISCUSSION THREAD

**Discussion by Ciprian167** [siliconcompiler/siliconcompiler#1168] [Help (Q&A)] [answered]
Hi. 

Please excuse my ignorance. I'm trying to better understand the entire open-source EDA landscape. I completely agree that people working with EDA needs to make use of build automation systems. The majority of companies use `make-based` systems or some custom specialized scripts, but that doesn't cut it. 

I am interested in your take on Bazel for EDA. Bazel is quite generic and can solve a lot of the issues SiliconCompiler is trying to tackle. Bazel has caching, remote caching, remote execution (to replace the old LSF), action graph to optimize the incremental builds. `Bzlmod` can be used for package management. Plus, Bazel has a larger user base. So, any optimization done in other development areas, will propagate to EDA world also. Sure, Bazel has a steep learning curve, but that is mostly paid by those who write the rules.

What are some things the SiliconCompiler is solving that Bazel does not and never will?

Thank you in advance!

**Comment by mithro** [comment]
FYI - Bazel support for EDA related topics can be found at https://github.com/hdl/bazel_rules_hdl and https://github.com/Lightelligence/rules_verilog

**Reply by mithro** [reply, marked as answer]
I would generally say that integrating the traditional proprietary EDA tools into Bazel is **_particularly_** hard. Bazel works best when "compiling the world from scratch" enabling very reproducible output. The less you look like a software compiler like Java / C++ the harder it is for your workload to map to Bazel.
